src/kit/ui/MultiSelectInput.js

```diff
@@ -15,7 +15,7 @@ export default class MultiSelectInput extends OverlayMixin(Component) {
   render ($$) {
     const selected = this.props.selected
     const isEmpty = selected.length === 0
-    const selectedLabels = selected.map(item => item.toString())
+    const selectedLabels = selected.map(item => this.context.api._renderEntity(item))
```

Comment:
I don't like this solution.
MultiSelectInput should be general. So we should rather use renderEntity in the specifc toString() implementation.

Reply:
I will revert this change.

Reply:
Also, it would be good to add a test to 'reveal' the problem before fixing it. This is a general request: Do not just fix things. Let the test-suite detect the problem. Then fix it.
In this case, I can just revert the change, but no effect on the test-suite, and then we are running in circles.

Reply:
@Integral look at my commits to learn how to approach this correctly.

Reply:
Another request: you named the change 'Show more information in affiliations field and popup', which is not what you have done in the change.
It would be better to say 'Use renderEntity() for MultiSelectInput options'.
This would make it very apparent that this is not a desired change.